test/unit/features/conftest.py

```diff
@@ -218,46 +216,27 @@ def prepare_mock_with_error(self, error: str):
                 json={'error': error},
             )
 
-    def runner(self, follower, mode, timeout=0, stop_after=0):
+    def runner(self, follower, mode, timeout=1, stop_after=0):
         """
         blocking runner with timeout
         """
 
-        class TimeoutError(Exception):
-            pass
-
-        def looper(changes, buf):
+        def main():
+            if mode == _Mode.LISTEN:
+                changes = follower.start()
+            elif mode == _Mode.FINITE:
+                changes = follower.start_one_off()
+            stop_timer = Timer(timeout, follower.stop)
```

Comment:
I think the function here might need a little more than `follower.stop`. For example, if the timer pops in a `Finite` case maybe we throw an assertion failure. I'm worrying about a case where we regress finite function and it doesn't cleanly terminate _by itself_ and the timer masks that from us.

Reply:
The behavior of the runner didn't change and in finite cases we generally have tests for both number of changes and exception, so if function doesn't terminate by itself we'll be getting assertion errors either on missing exception or count. We even had to suppress `TimeoutError` before to make threads timeout _not_ to mask exit status of the runner.

Reply:
Got it, thanks for clarifying.